noipy/main.py

```diff
@@ -34,6 +32,67 @@
     r'(?:/?|[/?]\S+)$', re.IGNORECASE)
 
 
+def auth_credentials(user_token, password, provider_class):
+    if provider_class.auth_type == 'T':
+        user_arg = user_token or utils.read_input(
+            "Paste your auth token: ")
+        auth = authinfo.ApiAuth(usertoken=user_arg)
+    else:
+        user_arg = user_token or utils.read_input(
+            "Type your username: ")
+        pass_arg = password or getpass.getpass("Type your password: ")
+        auth = authinfo.ApiAuth(user_arg, pass_arg)
+
+    return auth
+
+
+def execute_ddns_update(update_ddns, ip, hostname, auth,
```

Comment:
This functions has too many parameters and still has some *Cognitive Complexity*. 
```sh
49-51: Function `execute_ddns_update` has 9 arguments (exceeds 4 allowed). Consider refactoring. [structure]
49-73: Function `execute_ddns_update` has a Cognitive Complexity of 8 (exceeds 5 allowed). Consider refactoring. [structure]
```
Maybe try a different approach here.